manifests/0000_30_cluster-api_13_vap.yaml

```diff
@@ -0,0 +1,121 @@
+        ])
+    - name: check-authoritativeAPI-clusterapi
+      expression: 'object.status.authoritativeAPI == "ClusterAPI"'
+    - name: check-param-match
+      expression: 'object.metadata.name == params.metadata.name'
+
+  variables:
+    # label maps
+    - name: newLabels
+      expression: "object.metadata.?labels.hasValue() ? object.metadata.?labels.value()     : {}"
+    - name: oldLabels
+      expression: "oldObject.metadata.?labels.hasValue()   ? oldObject.metadata.?labels.value()   : {}"
+    - name: paramLabels
+      expression: "params.metadata.labels"
+
+    # annotation maps
+    - name: newAnn
+      expression: "object.metadata.?annotations.hasValue() ? object.metadata.?annotations.value() : {}"
+    - name: oldAnn
+      expression: "oldObject.metadata.?annotations.hasValue() ? oldObject.metadata.?annotations.value() : {}"
+
+    - name: specLockedExceptAuthoritativeAPI
+      expression: >
+        object.spec.authoritativeAPI != oldObject.spec.authoritativeAPI &&
+        [
+          [object.spec.?lifecycleHooks,   oldObject.spec.?lifecycleHooks],
+          [object.spec.?metadata,         oldObject.spec.?metadata],
+          [object.spec.?providerID,       oldObject.spec.?providerID],
+          [object.spec.?providerSpec,     oldObject.spec.?providerSpec],
+          [object.spec.?taints,           oldObject.spec.?taints]
+        ].all(p,
+          p[0].hasValue()
+            ? p[1].hasValue() && p[0].value() == p[1].value()
+            : !p[1].hasValue()
+        )
+
+  # All validations must evaluate to TRUE
+  validations:
+    # Only spec.authoritativeAPI may change
+    - expression: "object.spec == oldObject.spec || variables.specLockedExceptAuthoritativeAPI"
+      message:  "You may only modify spec.authoritativeAPI. Any other change inside .spec is not allowed."
+
+    # Guard machine.openshift.io/* and kubernetes.io/* labels
+    - expression: >
+        !(
+          variables.newLabels.exists(k,
+              (k.startsWith('machine.openshift.io') || k.startsWith('kubernetes.io')) &&
+              ((k in variables.oldLabels ? variables.oldLabels[k] : null) != variables.newLabels[k])
+          ) ||
+          variables.oldLabels.exists(k,
+              (k.startsWith('machine.openshift.io') || k.startsWith('kubernetes.io')) &&
+              !(k in variables.newLabels)
+          )
+        )
+      message: "Cannot modify or delete any machine.openshift.io/* or kubernetes.io/* label."
+
+    # Guard machine.openshift.io/* annotations
+    - expression: >
+        !(
+          variables.newAnn.exists(k,
+              k.startsWith('machine.openshift.io') &&
+              ((k in variables.oldAnn ? variables.oldAnn[k] : null) != variables.newAnn[k])
+          ) ||
+          variables.oldAnn.exists(k,
+              k.startsWith('machine.openshift.io') &&
+              !(k in variables.newAnn)
+          )
+        )
+      message: "Cannot modify or delete any machine.openshift.io/* annotation."
```

Comment:
Aren't there any k8s.io annotations?
IIRC we have the paused one for example. This is MAPI I know so that might not apply here, but might be worth checking

Reply:
Ah yep - I'll add this to my list.

I'm also not sure (from the card here) on what these exactly entail

```
Labels and Annotations that are not material to operation
  Are not important for operation of Cluster API or Machine API
```

Reply:
Not sure either, but probably best to not lock it down too much

Reply:
Yeah, I've not really covered this outside of cluster-x or machine.openshift.io labels /annotations.

cc @JoelSpeed I'm probably missing something